pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -252,4 +252,10 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.EnableDebugFlagsHandler == nil {
 		obj.EnableDebugFlagsHandler = utilpointer.BoolPtr(true)
 	}
+	if obj.ShutdownGracePeriod == zeroDuration {
+		obj.ShutdownGracePeriod = metav1.Duration{Duration: 0}
```

Comment:
Why not set the default value to turn it on, is wait for GA to set it?

Reply:
We had this discussion on [slack thread here](https://kubernetes.slack.com/archives/C0BP8PW9G/p1614888726200500). 

The idea is that different users might have different requirements for what `ShutdownGracePeriod` / `ShutdownGracePeriodCriticalPods` is acceptable for their environments, so it's difficult to come up with a general value. As a result, we decided to just default it zero (but enable the feature gate). So if users want to use this feature, they'll only need to configure kubelet config with `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods`